Feat : 7단계 소단계 작성

Jihye-kr opened the pull request:
## 📌 Issue

<!-- 해결하려는 이슈 번호나 주제를 명확하게 적어주세요. -->

- 관련 이슈: close #219 

---

## 🛠 작업 내용

1. 7단계 체크리스트 기능 구현

- 7-1단계: 명의도용 대출 사기 예방 체크리스트 (3개 문항)
- 7-2단계: 브로커를 통한 전세보증 사기 예방 체크리스트 (3개 문항)
- 각 문항별 예/아니오 선택 및 피드백 메시지 제공

2. CombinedContent 타입 지원 개선

- ModalContent에서 CombinedContent 타입 렌더링 시 Swiper 적용
- TextOnly와 RadioGroup 섹션을 별도 슬라이드로 분리하여 표시
- 슬라이더 기반 페이지 네비게이션 및 페이지 인디케이터 추가

3. RadioGroup 컴포넌트 로직 개선

- CombinedContent 타입에서 실제 질문 제목 추출 로직 수정
- allAnswered 계산 정확도 향상으로 결과 표시 조건 개선
- successMessages가 없을 때 messagesContainer 숨김 처리

4. 데이터베이스 및 API 연동

- Step 7-2 테이블 추가 (ID: 20)

5. 회원가입, 로그인 refactoring

- 비밀번호, pin 번호 보기 버튼 클릭 시에만 번호 노출
- 회원가입 성공 , 에러시 modal 보이게

## 🚀 기타 사항

<!-- 리뷰어가 추가적으로 알아야 할 사항이 있다면 기재해주세요. -->
<img width="200" height="871" alt="스크린샷 2025-08-23 오후 10 54 45" src="https://github.com/user-attachments/assets/9963c8c5-f178-4d0b-9950-a00a2b20340c" />

<img width="200" height="868" alt="스크린샷 2025-08-23 오후 10 55 23" src="https://github.com/user-attachments/assets/3212a4ad-97dd-4cfd-ad75-234e2af621ad" />

<img width="200" height="870" alt="스크린샷 2025-08-23 오후 10 55 51" src="https://github.com/user-attachments/assets/392c259f-130c-4149-8c33-85873d07432a" />

<img width="200" height="872" alt="스크린샷 2025-08-23 오후 10 56 15" src="https://github.com/user-attachments/assets/daf9802d-399e-4e92-a1f5-1a4ecaf68c7f" />

- [x] build 체크 했나요?

- [x] dev를 pull 받은 뒤 merge 했나요?

Jihye-kr committed:
refactor: CombinedContent 컴포넌트의 인라인 스타일을 styles.ts 파일로 분리

Jihye-kr committed:
feat : Table 데이터타입 추가 작성(@layout-SY)

Jihye-kr committed:
fix : 소단계 닫을 때 중단계 페이지 초기화 방지

Jihye-kr committed:
refactor: CombinedContent 컴포넌트에서 any 타입 제거 및 타입 안전성 개선

Jihye-kr committed:
Merge branch 'develop' into feat/#219

Jihye-kr committed:
소단계-7단계 작성 중

Jihye-kr committed:
Resolve merge conflicts and integrate develop branch features

- Resolved merge conflicts in types.ts, GeneralPage.tsx, and other files
- Integrated develop branch features including:
  - JeonseGuarantee components and functionality
  - RealEstate data processing and risk assessment
  - PDF viewer and original document handling
  - Enhanced modal overlays and styling
  - New API endpoints and backend functionality
  - Improved TypeScript interfaces with successMessages and summary fields
  - Modern styling and component architecture

Jihye-kr committed:
feat : 7-1단계 작성완료

Jihye-kr committed:
feat: 7단계 체크리스트 기능 구현 및 CombinedContent 타입 지원 개선

- 7-1단계: 명의도용 대출 사기 예방 체크리스트 구현
- 7-2단계: 브로커를 통한 전세보증 사기 예방 체크리스트 구현
- CombinedContent 타입 지원 개선
  - ModalContent에서 CombinedContent 타입 렌더링 시 Swiper 적용
  - 각 섹션을 별도 슬라이드로 표시하여 사용자 경험 향상
  - 결과 메시지와 유용한 링크 표시 기능 추가

- RadioGroup 컴포넌트 개선
  - CombinedContent 타입에서 실제 질문 제목 추출 로직 수정
  - allAnswered 계산 정확도 향상
  - successMessages가 없을 때 messagesContainer 숨김 처리

- 데이터베이스 스키마 업데이트
  - Step 7-2 테이블 추가 (ID: 20)
  - 7단계 체크리스트 데이터 저장/로드 기능 완성

- UI/UX 개선
  - 슬라이더 기반 페이지 네비게이션
  - 페이지 인디케이터 추가
  - 일관된 스타일링 적용

icw0201 commented:
1. 단계별 캡처 이미지 첨부 요망
2. PR에 작성해주신 `라디오 버튼 선택 결과를 DB에 저장/로드하는 기능 완성` 부분 이미 되어 있지 않았나요?

icw0201 commented on app/(anon)/@detail/steps/[step-number]/[detail]/_components/ModalContent.tsx:
```diff
@@ -138,6 +129,99 @@ export default function ModalContent() {
+                {section.type === 'DataGrid' && (
+                  <DataGrid
+                    data={section.data as unknown as { left: string; right?: string }[]}
+                  />
```
체크리스트는 왜 type에 없어염?

icw0201 commented on app/(anon)/@detail/steps/[step-number]/[detail]/_components/contents/RadioGroup.tsx:
```diff
@@ -260,20 +320,32 @@ const RadioGroup = ({ data }: RadioGroupProps) => {
+                {selectedAnswers[sectionIndex] === 'no' && section.successMessages && (
+                  // '아니오' 선택 시 안전 메시지
+                  section.successMessages.map((message, messageIndex) => (
+                    <div key={messageIndex} className={styles.messageItem}>{message}</div>
```
이거 수정하신 뒤에 제 1-2 페이지에 변동 안생기나요?

Jihye-kr replied:
네!

icw0201 commented on app/(anon)/@detail/steps/[step-number]/[detail]/_components/contents/RadioGroup.tsx:
```diff
@@ -260,20 +320,32 @@ const RadioGroup = ({ data }: RadioGroupProps) => {
+                )}
+                {section.link && (
+                  <div className={styles.linkContainer}>
+                    <a href={section.link} target="_blank" rel="noopener noreferrer" className={styles.link}>
```
Link?

Jihye-kr replied:
수정완!

Jihye-kr committed:
fix: any 타입 에러 수정으로 Vercel 빌드 실패 해결

- steps/[step-number]/page.tsx: any 타입을 FlipBookInstance 인터페이스로 변경
- RadioGroup.tsx: any 타입 제거
- TypeScript 컴파일 에러 해결로 빌드 성공

Jihye-kr committed:
feat: Checklist 타입 지원 추가 및 타입 에러 수정

- types.ts: ChecklistSection 인터페이스 추가
- ModalContent.tsx: Checklist 타입 렌더링 지원 추가
- CombinedContent.tsx: Checklist 타입 처리 및 타입 에러 수정
- RadioGroup.tsx: Link 관련 기능 유지

icw0201 commented on app/(anon)/@detail/steps/[step-number]/[detail]/_components/contents/CombinedContent.tsx:
```diff
@@ -40,9 +40,12 @@ const CombinedContent = ({
         content = <DataGrid data={section.data} />;
         break;
+      case 'Checklist':
+        content = <RadioGroup data={section.data} />;
```
이름 CheckListGroup 인데여
그리고 content왜 라디오그룹이죠

Jihye-kr replied:
삭제완..;;

icw0201 reviewed (changes requested):
냐...

Jihye-kr committed:
refactor: Checklist 타입 제거 및 Next.js Link 통일

- types.ts: ChecklistSection 인터페이스 제거, RadioGroup으로 통일
- ModalContent.tsx: Checklist 케이스 제거
- CombinedContent.tsx: Checklist 케이스 제거
- RadioGroup.tsx: 모든 링크를 Next.js Link 컴포넌트로 통일
- 코드 중복 제거 및 일관성 향상

Jihye-kr committed:
feat : Link import

Jihye-kr committed:
feat: Add CheckListGroup to CombinedContent and resolve merge conflicts

- Add CheckListGroup component to CombinedContent handling in ModalContent
- Update types.ts to include CheckListGroupSection interface
- Resolve merge conflicts in RadioGroup.tsx by using proper types
- Update CheckListGroup.tsx to use types from types file
- Ensure type safety across all content components

Jihye-kr committed:
Merge branch 'develop' of https://github.com/FRONT-END-BOOTCAMP-PLUS-5/BoGam into develop

Jihye-kr committed:
Merge branch 'develop' into feat/#219

icw0201 commented:
<img width="445" height="149" alt="image" src="https://github.com/user-attachments/assets/d8f7d449-1363-43f6-9d42-71649829f654" />

이거 제가 구현한 부분이라서, 하신 부분이 어디인지 좀 더 상세하게 작성해주신다면 감사할 것 같습니다

icw0201 reviewed (approved):
APP

Jihye-kr committed:
feat: OTP 입력 필드에 비밀번호 보기 기능 추가 및 회원가입 완료 모달 구현

- OTP 입력 필드에 눈 아이콘 버튼 추가 (누르고 있을 때만 비밀번호 표시)
- PasswordInput 컴포넌트 수정 (눈 아이콘 단순화)
- 회원가입 완료 시 alert 대신 ConfirmModal 사용

Jihye-kr committed:
feat: 회원가입 에러 처리 개선 및 ConfirmModal 버튼 조건부 렌더링

- 회원가입 에러 발생 시 사용자에게 모달로 에러 메시지 표시
- ConfirmModal에서 cancelText가 없으면 취소 버튼 숨김
- 에러 모달은 확인 버튼만 표시하여 UI 단순화

aiden0413 committed:
Feat: 메인페이지 스타일 개선, 리팩토링 /#245 (#255)

* feat: ConfirmModal을 DaumPostcodeModal에 통합 및 AuthLanding 컴포넌트 제거

* feat: 주소 드롭다운 UI/UX 개선 및 실거래가 페이지네이션 추가

* fix: 빌드경고 해결

* feat: 사용자 이름 및 닉네임 골드색으로 변경

layout-SY committed:
refactor : 등기부등본, 납세증명서, 부동산 중개업자 컴포넌트의 input/outut 구현 (#253)

* refactor : 주소 데이터 상태 관리 방식 변경

* refactor : fetch 최적화 및 파일 위치 변경, utils로 묶기

* feat : 소분류 페이지에 y축 스크롤 추가

* feat : 원문 보기 컴포넌트 구현

* refactor : 폴더 구조 변경

* refactor : 컴포넌트에 있던 hook 관련 로직 분리

* refactor : 컴포넌트 단위별 분리

* feat : 위험도 측정 컴포넌트 구현

* feat : 5-2단계의 등기부등본 위험도 측정 기능 구현

* feat : 등기부등본 JSON 데이터를 완전히 분해 하여 검색하는 페이지 구현

* refactor : 빌드 오류 해결

* design : 공용 색 제거

* feat : 납세증명서 소분류 페이지에 연결

* feat : 스크롤바 추가 및 단계에 따른 서류 조회 컴포넌트 분기 리팩토링

* feat : 부동산 중개업자 관련 컴포넌트 구현

* feat : 부동산 중개업자 위험도 측정 로직 추가

* design : "입력", "결과" 컴포넌트 스타일 추가

* feat : 컨텐츠 분기 컴포넌트 구현

* feat : 서류 관련 컴포넌트에서 사용하는 input Form 컴포넌트 별도 생성

* feat : 추가된 위험도 측정 로직에 따른 컴포넌트 기능 추가 구현

* refactor : 변경된 구조에 따른 등기부등본 컴포넌트 수정

* feat : 납세증명서 컴포넌트 위험도 측정 추가 및 UI/UX 수정

* feat : 부동산 중개업자 위험도 측정 로직 추가 및 API 연결

Jihye-kr committed:
Fix TaxCertResultDisplay useEffect and type issues

Jihye-kr committed:
fix : 빌드 오류 해결

Jihye-kr committed:
Merge branch 'develop' into feat/#219
